models/topicsModel.js

```diff
@@ -1,16 +1,25 @@
 const db = require("../db/connection");
-const fs = require("fs/promises")
+const fs = require("fs/promises");
 
+exports.getAllTopics = () => {
+  return db.query("SELECT * FROM topics").then((topics) => {
+    return topics.rows;
+  });
+};
 
-exports.getAllTopics = () =>{
-    return db.query("SELECT * FROM topics").then((topics)=>{
-        return topics.rows
-    })
-}
+exports.getAllApis = () => {
+  return fs.readFile("./endpoints.json", "utf-8").then((apis) => {
+    return JSON.parse(apis);
+  });
+};
 
-exports.getAllApis = () =>{
-    return fs.readFile("./endpoints.json", "utf-8").then((apis)=>{
-        return JSON.parse(apis)
-    })
-    
-}
\ No newline at end of file
+exports.getArticleById = (id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1`, [id])
+    .then((article) => {
+      if (!article) {
+        return Promise.reject({ status: 400, msg: "bad request" });
+      }
```

Comment:
This if statement will only be triggered if nothing is sent back from the query. This would only happen if there was an error, so it would go directly to the catch block. This means this if block will actually never be triggered.

You will need to have something similar for this for a test to check for valid but non-existent ids